src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/http_action/config/PerformanceAnalyzerOverridesClusterConfigAction.java

```diff
@@ -0,0 +1,176 @@
+            throws IOException {
+        RestChannelConsumer consumer;
+        if (request.method() == RestRequest.Method.GET) {
+            consumer = handleGet();
+        } else if (request.method() == RestRequest.Method.POST) {
+            consumer = handlePost(request);
+        } else {
+            String reason = "Unsupported method:" + request.method().toString() + " Supported: [GET, POST]";
+            consumer = sendErrorResponse(reason, RestStatus.METHOD_NOT_ALLOWED);
+        }
+
+        return consumer;
+    }
+
+    /**
+     * Handler for the GET method.
+     *
+     * @return RestChannelConsumer that sends the current config overrides when run.
+     */
+    private RestChannelConsumer handleGet() {
+        return channel -> {
+            try {
+                final ConfigOverrides overrides = overridesWrapper.getCurrentClusterConfigOverrides();
+                XContentBuilder builder = channel.newBuilder();
+                builder.startObject();
+                builder.field(OVERRIDES_FIELD, ConfigOverridesHelper.serialize(overrides));
+                builder.endObject();
+                channel.sendResponse(new BytesRestResponse(RestStatus.OK, builder));
+            } catch (IOException ioe) {
+                LOG.error("Error sending response", ioe);
+            }
+        };
+    }
+
+    /**
+     * Handler for the POST method.
+     *
+     * @param request The POST request.
+     * @return RestChannelConsumer that updates the cluster setting with the requested config
+     * overrides when run.
+     * @throws IOException if an exception occurs trying to parse or execute the request.
+     */
+    private RestChannelConsumer handlePost(final RestRequest request) throws IOException {
+        String jsonString = XContentHelper.convertToJson(request.content(), false, XContentType.JSON);
+        ConfigOverrides requestedOverrides = ConfigOverridesHelper.deserialize(jsonString);
+
+        if (!validateOverrides(requestedOverrides)) {
+            String reason = "enable set and disable set should be disjoint";
+            return sendErrorResponse(reason, RestStatus.BAD_REQUEST);
+        }
+
+        configOverridesClusterSettingHandler.updateConfigOverrides(requestedOverrides);
+        return channel -> {
+            XContentBuilder builder = channel.newBuilder();
+            builder.startObject();
+            builder.field(OVERRIDE_TRIGGERED_FIELD, true);
+            builder.endObject();
+            channel.sendResponse(new BytesRestResponse(RestStatus.OK, builder));
+        };
+    }
+
+    private boolean validateOverrides(final ConfigOverrides requestedOverrides) {
+        boolean isValid = true;
+
+        // Check if we have both enable and disable components
+        if (requestedOverrides.getDisable() == null || requestedOverrides.getEnable() == null) {
+            return true;
+        }
+
+        // Check if any RCA nodes are present in both enabled and disabled lists.
```

Comment:
We  should also check that the values provided are string type.

Reply:
The deserialize fails if the provided values are not string types.